src/debug.c

```diff
@@ -1123,73 +1123,88 @@ void bugReportStart(void) {
 }
 
 #ifdef HAVE_BACKTRACE
-static void *getMcontextEip(ucontext_t *uc) {
+
+/* Returns the current eip and set it to the given new value (if its not NULL) */
+static void* getAndSetMcontextEip(ucontext_t *uc, void *eip) {
 #define NOT_SUPPORTED() do {\
     UNUSED(uc);\
+    UNUSED(eip);\
     return NULL;\
 } while(0)
+#define GET_SET_VALUD_AT(add, val) do {\
```

Comment:
```suggestion
#define GET_SET_RETURN(target_var, new_val) do {\
```

seeing a function that's called "get" is fine, but a macro that's called GET, and actually returns (changes the flow) is misleading IMHO.